src/beignet/func/_implicit_diff_root_scalar.py

```diff
@@ -33,27 +34,33 @@ def backward(ctx, *grad_outputs):
                 # optimality condition:
                 # f(x^*(theta), theta) = 0
 
-                A, *B = torch.func.jacrev(f, argnums=tuple(range(nargs + 1)))(
-                    xstar, *args
-                )
-
-                if A.ndim == 0:
-                    return tuple(
-                        -g * b / A for g, b in zip(grad_outputs, B, strict=True)
-                    )
-                elif A.ndim == 2:
-                    # NOTE even in this case A is diagonal because f is scalar
-                    return tuple(
-                        torch.linalg.solve(A, -g * b)
-                        for g, b in zip(grad_outputs, B, strict=True)
-                    )
-                else:
-                    raise RuntimeError(f"{A.ndim=} != 0 or 2")
+                argnums = tuple(range(nargs + 1))
+
+                a, *b = torch.func.jacrev(f, argnums=argnums)(xstar, *args)
+
+                match a.ndim:
+                    case 0:
+                        output = ()
+
+                        for g, b2 in zip(grad_outputs, b, strict=True):
+                            output = (*output, -g * b2 / a)
```

Comment:
why is this style preferred? I don't think the original expression with the generator introduces any mutation right?